src/bin/pg_autoctl/monitor.c

```diff
@@ -3075,6 +3125,128 @@ monitor_get_notifications(Monitor *monitor)
 }
 
 
+/*
+ * monitor_wait_for_state_change waits for timeout milliseconds or until we
+ * receive a notification for a state change concerning the given nodeId,
```

Comment:
There's no `nodeId` parameter.
```suggestion
 * receive a notification for a state change concerning the given groupId,
```